src/extension/background-script/connectors/alby.ts

```diff
@@ -0,0 +1,378 @@
+    });
+
+    return authResult;
+  }
+
+  async getAccessToken(authResult: string, authClient: auth.OAuth2User) {
+    const authToken = new URL(authResult).searchParams.get("code");
+    if (!authToken) {
+      throw new Error("Authentication failed: missing token");
+    }
+
+    await authClient.requestAccessToken(authToken);
+  }
+
+  generateHmacVerification(uri: string) {
+    const mac = hmacSHA256(uri, HMAC_VERIFY_HEADER_KEY).toString(Base64);
+    return encodeURIComponent(mac);
+  }
+
+  async request<Type>(
+    method: Method,
+    path: string,
+    args?: Record<string, unknown>
+  ): Promise<Type> {
+    throw new Error(
+      "Unsupported. Migrate to alby-js-sdk and remove this function"
+    );
+  }
+
+  private async _getAlbyClient(): Promise<Client> {
+    if (this._clientPromise) {
+      return this._clientPromise;
+    }
+
+    this._clientPromise = new Promise<Client>((resolve, reject) => {
+      (async () => {
+        try {
+          this._authUser = await this.authorize();
+          resolve(new Client(this._authUser, this._getRequestOptions()));
+        } catch (error) {
+          reject(error);
+        }
+      })();
+    });
+    return this._clientPromise;
+  }
+
+  private async _request<T>(func: (client: Client) => T) {
+    const client = await this._getAlbyClient();
+    if (!this._authUser) {
+      throw new Error("this._authUser was not created");
+    }
+    const oldToken = this._authUser?.token;
+    let result: T;
+    try {
+      result = await func(client);
+    } catch (error) {
+      console.error(error);
+      throw error;
+    } finally {
+      const newToken = this._authUser.token;
+      if (newToken && newToken !== oldToken) {
+        await this._updateOAuthToken(newToken);
+      }
+    }
+    return result;
+  }
+
+  private _getRequestOptions(): Partial<RequestOptions> {
+    return {
+      ...(process.env.ALBY_API_URL
+        ? { base_url: process.env.ALBY_API_URL }
+        : {}),
+    };
+  }
+
+  private async _updateOAuthToken(newToken: Token) {
+    console.info("Updating account oauth token", newToken);
+    const access_token = newToken.access_token;
+    const refresh_token = newToken.refresh_token;
+    const expires_at = newToken.expires_at;
+
+    if (access_token && refresh_token && expires_at) {
+      this.config.oAuthToken = { access_token, refresh_token, expires_at };
+      if (this.account.id) {
+        const accounts = state.getState().accounts;
+        const password = (await state.getState().password()) as string;
+
+        const configData = decryptData(
+          accounts[this.account.id].config,
+          password
+        );
+        configData.oAuthToken = this.config.oAuthToken;
+        accounts[this.account.id].config = encryptData(configData, password);
+        state.setState({ accounts });
+        // make sure we immediately persist the updated accounts
+        await state.getState().saveToStorage();
+      }
+      console.info(
```

Comment:
review and remove console logs that leak oauth / refresh token

Reply:
removed